src/scores/continuous/standard_impl.py

```diff
@@ -382,3 +383,75 @@ def multiplicative_bias(
     fcst, obs = broadcast_and_match_nan(fcst, obs)
     multi_bias = fcst.mean(dim=reduce_dims) / obs.mean(dim=reduce_dims)
     return multi_bias
+
+
+# NSE code
+def nse(fcst, obs, reduce_dims=None, preserve_dims=None, weights=None, angular=False):
+    """
+    Calculate Nash-Sutcliffe Efficiency (NSE) for forecast and observed data.
+
+    Args:
+        fcst (FlexibleArrayType or list): Forecast or predicted variables.
+        obs (FlexibleArrayType or list): Observed variables.
+        reduce_dims (FlexibleDimensionTypes, optional): Dimensions to reduce along. Defaults to None.
+        preserve_dims (FlexibleDimensionTypes, optional): Dimensions to preserve. Defaults to None.
+        weights (xr.DataArray, optional): Weights to apply to error calculation. Defaults to None.
+        angular (bool, optional): Whether to treat data as angular (circular). Defaults to False.
+
+    Returns:
+        FlexibleArrayType: Nash-Sutcliffe Efficiency (NSE) value.
+
+    Raises:
+        ValueError: If the input arrays are of different lengths or incompatible types.
+
+    References:
+        - references
+        - https://en.wikipedia.org/wiki/Nash–Sutcliffe_model_efficiency_coefficient
+        - https://hess.copernicus.org/articles/26/4801/2022/
+
+    Examples:
+        # Case 1: Xarray DataArray
+        >>> fcst_xr = xr.DataArray([3, 4, 5, 6, 7])
+        >>> obs_xr = xr.DataArray([2, 3, 4, 5, 6])
+        >>> nse(fcst_xr, obs_xr)
+        0.50
+
+    """
+    # Convert datasets to data arrays if needed
+    if isinstance(fcst, xr.Dataset):
+        data_variable_name = list(fcst.data_vars.keys())[0]  # Get the name of the first data variable
+        fcst = fcst.to_array(dim=data_variable_name)
```

Comment:
I don't think that we want to convert xr.Datasets to xr.DataArrays. I think this will only take out the first data_var anyway. 

The code needs to be updated to work on Datasets with all data_vars

Reply:
I ran across a similar thing in another area of scores, whereby the calculation was really only valid on a DataArray. In that case, it would be possible to split out each variable from a DataSet and then do the calculation on each variable, then reconstruct the results. Is the same kind of thing going on here?